#0: Add CMake build flag ENABLE_LIBCXX to selectively enable/disable libc++

nsmithtt opened the pull request:
This PR enables users of ttnn/ttmetal to disable libc++ if they want to link against libstdc++ instead.

TT-billteng commented on tt_metal/tt_stl/aligned_allocator.hpp:
```diff
@@ -43,7 +43,7 @@ class aligned_allocator {
     }
 
-    void deallocate(T* p, std::size_t size) const noexcept { ::operator delete[](p, size * sizeof(T), alignment); }
+    void deallocate(T* p, std::size_t size) const noexcept { ::operator delete[](p, alignment); }
```
why is this change needed?

nsmithtt replied:
It wasn't present in libstdc++'s header, but after a bit more digging it seems like it can be enabled if we compile with flag ` -fsized-deallocation`.  In reading cppref I can't tell that there's any difference between APIs 3-4 and 7-8. https://en.cppreference.com/w/cpp/memory/new/operator_delete

nsmithtt replied:
I can update the PR to use `-fsized-deallocation` if that seems better.

marty1885 commented on CMakeLists.txt:
```diff
@@ -72,6 +72,8 @@ option(ENABLE_CODE_TIMERS "Enable code timers" OFF)
 
 option(ENABLE_TRACY "Enable Tracy Profiling" OFF)
 
+option(ENABLE_LIBCXX "Disable using libc++" ON)
```
This comment is confusing. The variable name indicates setting this variable to ON means to enable lib++ but comment says **disable**

nsmithtt replied:
Thanks! I renamed the variable and not the comment :)

nicholashaydensmith committed:
#0: Add CMake build flag ENABLE_LIBCXX to selectively enable/disable libc++

This PR enables users of ttnn/ttmetal to disable libc++ if they want to link against libstdc++ instead.
